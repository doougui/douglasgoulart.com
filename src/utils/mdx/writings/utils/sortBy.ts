import { Writing } from 'types/Writing';
import { parseToDate } from 'utils/time/parseToDate';
import { SortTypesKeys } from '../types';

function byNew(data: Writing[]) {
  return data.sort((a, b) => {
    const aAsDate = parseToDate(a.publishedAt);
    const bAsDate = parseToDate(b.publishedAt);
    return Number(bAsDate) - Number(aAsDate);
  });
}

function byName(data: Writing[]) {
  return data.sort((a, b) => (a.title > b.title ? 1 : -1));
}

export function sortBy(data: Writing[], type: SortTypesKeys): Writing[] {
  const sortTypes: { [key in SortTypesKeys]: () => Writing[] } = {
    new: () => byNew(data),
    name: () => byName(data),
  };

  return sortTypes[type]();
}
