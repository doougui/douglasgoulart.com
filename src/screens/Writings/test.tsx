import { render, screen } from 'utils/tests';

import { Writings } from '.';

// eslint-disable-next-line @typescript-eslint/no-var-requires
const useRouter = jest.spyOn(require('next/router'), 'useRouter');

useRouter.mockImplementation(() => ({
  push: jest.fn(),
  query: '',
  pathname: '/',
  route: '/',
}));

describe('<Writings />', () => {
  it('should render correctly', () => {
    render(<Writings />);

    expect(
      screen.getByRole('heading', { name: /Writings/i }),
    ).toBeInTheDocument();
  });
});
