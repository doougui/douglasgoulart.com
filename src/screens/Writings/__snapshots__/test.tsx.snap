// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Writings /> should render heading, filter and writings 1`] = `
.c22 {
  font-size: 3.2rem;
  line-height: 4.2rem;
  color: #EDEDEF;
  font-weight: 600;
}

.c23 {
  font-size: 1.6rem;
  line-height: 2.4rem;
  color: #888888;
}

.c1 {
  width: 100%;
  max-width: 94rem;
  margin-left: auto;
  margin-right: auto;
  padding-left: calc(4rem / 2);
  padding-right: calc(4rem / 2);
}

.c38 {
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  font-size: 1.6rem;
  line-height: 2.4rem;
  color: #EDEDEF;
  background-color: #107BEB;
  border-radius: 0.4rem;
  padding: 1rem 4rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border: none;
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c27 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 1.5rem;
  color: #EDEDEF;
  font-size: 1.6rem;
}

.c28 {
  font-family: Inter,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  font-weight: 600;
}

.c36 {
  all: unset;
  width: 100%;
  font-size: 1.6rem;
  background: #2D2D2F;
  padding: 1.2rem 1.5rem;
  border-radius: 0.4rem;
  color: #EDEDEF;
}

.c36::-webkit-input-placeholder {
  color: #888888;
}

.c36::-moz-placeholder {
  color: #888888;
}

.c36:-ms-input-placeholder {
  color: #888888;
}

.c36::placeholder {
  color: #888888;
}

.c26 {
  margin-top: 5rem;
  padding-top: 3rem;
  border-top: 1px solid #2D2D2F;
}

.c29 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  gap: 8rem;
  margin-top: 2.5rem;
}

.c30 {
  color: #EDEDEF;
  font-size: 1.6rem;
  line-height: 2.4rem;
  font-weight: 400;
}

.c34 {
  font-size: 1.6rem;
  color: #888888;
}

.c31 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 1.5rem;
  margin-top: 1.5rem;
}

.c32 {
  color: #888888;
  font-size: 1.6rem;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c32:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c39 {
  color: #EDEDEF;
  font-size: 1.6rem;
  line-height: 2.4rem;
  margin-top: 3rem;
}

.c33 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c35 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-top: 1rem;
}

.c35 button {
  border-top-left-radius: 0;
  border-bottom-left-radius: 0;
  padding-left: 1.5rem;
  padding-right: 1.5rem;
}

.c37 {
  border-top-right-radius: 0;
  border-bottom-right-radius: 0;
  min-width: 10rem;
}

.c4 {
  font-family: Poppins;
  font-size: 3.2rem;
  color: #EDEDEF;
}

.c5 {
  color: #107BEB;
}

.c3 {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c6 {
  display: none;
}

.c20 {
  width: 3rem;
  height: 3rem;
  padding: 5px;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 0.4rem;
  background: #2D2D2F;
  -webkit-transition: background-color 0.1s ease-in-out;
  transition: background-color 0.1s ease-in-out;
}

.c20:hover {
  background: #1C1C1C;
}

.c19 {
  width: 3rem;
  height: 3rem;
  padding: 5px;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 0.4rem;
  background: #2D2D2F;
  -webkit-transition: background-color 0.1s ease-in-out;
  transition: background-color 0.1s ease-in-out;
  background: #504F57;
}

.c19:hover {
  background: #1C1C1C;
}

.c18 {
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
  border-radius: 0.4rem;
  background: #2D2D2F;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 5px;
  gap: 5px;
}

.c10 {
  padding-bottom: 2rem;
  padding-top: 2rem;
  border-bottom: 1px solid #504F57;
}

.c11 {
  color: #EDEDEF;
  font-size: 1.6rem;
  line-height: 1.8rem;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c9 {
  width: 100%;
  list-style: none;
  margin-top: 10rem;
  text-align: right;
}

.c8 {
  width: 100vw;
  height: 100vh;
  background: #151718;
  position: fixed;
  left: 0;
  top: 0;
  z-index: 20;
  padding: 0 calc(4rem / 2);
  pointer-events: none;
}

.c13 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 4px;
}

.c12 {
  all: unset;
  position: fixed;
  right: calc(4rem / 2);
  z-index: 50;
}

.c14 {
  width: 2.7rem;
  height: 3px;
  border-radius: 2px;
  background-color: white;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c15 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c16 {
  font-size: 1.6rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #EDEDEF;
  padding: 0 1.5rem;
  color: #107BEB;
}

.c17 {
  font-size: 1.6rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #EDEDEF;
  padding: 0 1.5rem;
}

.c0 {
  margin: 5rem 0;
}

.c21 {
  margin-top: 10rem;
}

.c24 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 1rem;
}

.c25 {
  margin-top: 3rem;
}

@media (max-width:768px) {
  .c6 {
    display: block;
  }
}

@media (min-width:768px) {
  .c7 {
    display: none;
  }
}

@media (max-width:768px) {
  .c15 {
    display: none;
  }
}

@media (max-width:768px) {
  .c0 {
    margin: 3rem 0;
  }
}

@media (max-width:768px) {
  .c21 {
    margin-top: 5rem;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <menu
      class="c2"
    >
      <a
        class="c3"
        href="/"
      >
        <div
          class="c4"
          data-testid="logo"
        >
          Douglas
          <span
            class="c5"
          >
            .
          </span>
        </div>
      </a>
      <div
        class="c6"
        data-testid="mobile-menu"
      >
        <div
          class="c7"
        >
          <nav
            aria-hidden="true"
            class="c8"
            data-testid="container"
            style="opacity: 0;"
          >
            <ul
              class="c9"
            >
              <li
                class="c10"
                style="opacity: 0; transform: translateX(200px) translateZ(0);"
              >
                <a
                  class="c11"
                  href="/"
                >
                  Home
                </a>
              </li>
              <li
                class="c10"
                style="opacity: 0; transform: translateX(200px) translateZ(0);"
              >
                <a
                  class="c11"
                  href="/about"
                >
                  About
                </a>
              </li>
              <li
                class="c10"
                style="opacity: 0; transform: translateX(200px) translateZ(0);"
              >
                <a
                  class="c11"
                  href="/writings"
                >
                  Writings
                </a>
              </li>
              <li
                class="c10"
                style="opacity: 0; transform: translateX(200px) translateZ(0);"
              >
                <a
                  class="c11"
                  href="/guestbook"
                >
                  Guestbook
                </a>
              </li>
            </ul>
          </nav>
          <button
            class="c12"
            type="button"
          >
            <div
              class="c13"
            >
              <span
                class="c14"
              />
              <span
                class="c14"
              />
            </div>
          </button>
        </div>
      </div>
      <nav
        class="c15"
        data-testid="desktop-menu"
      >
        <a
          class="c16"
          href="/"
        >
          Home
        </a>
        <a
          class="c17"
          href="/about"
        >
          About
        </a>
        <a
          class="c17"
          href="/writings"
        >
          Writings
        </a>
        <a
          class="c17"
          href="/guestbook"
        >
          Guestbook
        </a>
        <div
          class="c18"
        >
          <div
            class="c19"
            data-testid="light-toggle"
          >
            <svg
              color="#EDEDEF"
              fill="currentColor"
              height="15"
              stroke="currentColor"
              stroke-width="0"
              style="color: rgb(237, 237, 239);"
              viewBox="0 0 24 24"
              width="15"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g>
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
                <path
                  d="M12 18a6 6 0 1 1 0-12 6 6 0 0 1 0 12zm0-2a4 4 0 1 0 0-8 4 4 0 0 0 0 8zM11 1h2v3h-2V1zm0 19h2v3h-2v-3zM3.515 4.929l1.414-1.414L7.05 5.636 5.636 7.05 3.515 4.93zM16.95 18.364l1.414-1.414 2.121 2.121-1.414 1.414-2.121-2.121zm2.121-14.85l1.414 1.415-2.121 2.121-1.414-1.414 2.121-2.121zM5.636 16.95l1.414 1.414-2.121 2.121-1.414-1.414 2.121-2.121zM23 11v2h-3v-2h3zM4 11v2H1v-2h3z"
                  fill-rule="nonzero"
                />
              </g>
            </svg>
          </div>
          <div
            class="c20"
            data-testid="dark-toggle"
          >
            <svg
              color="#EDEDEF"
              fill="currentColor"
              height="15"
              stroke="currentColor"
              stroke-width="0"
              style="color: rgb(237, 237, 239);"
              viewBox="0 0 24 24"
              width="15"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g>
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
                <path
                  d="M10 7a7 7 0 0 0 12 4.9v.1c0 5.523-4.477 10-10 10S2 17.523 2 12 6.477 2 12 2h.1A6.979 6.979 0 0 0 10 7zm-6 5a8 8 0 0 0 15.062 3.762A9 9 0 0 1 8.238 4.938 7.999 7.999 0 0 0 4 12z"
                  fill-rule="nonzero"
                />
              </g>
            </svg>
          </div>
        </div>
      </nav>
    </menu>
    <main
      class="c21"
    >
      <h1
        class="c22"
      >
        Writings
      </h1>
      <p
        class="c23"
      >
        Brain dumps, explorations, and how-to guides. I write about different topics, such as design system, accessibility, start ups and user/dev experience.
      </p>
      <div
        data-testid="filter"
      />
      <div
        class="c24 c25"
        spacing="1rem"
      >
        <div
          data-testid="writing"
        />
        <div
          data-testid="writing"
        />
        <div
          data-testid="writing"
        />
        <div
          data-testid="writing"
        />
        <div
          data-testid="writing"
        />
        <div
          data-testid="writing"
        />
      </div>
    </main>
    <footer
      class="c26"
    >
      <div
        class="c27"
      >
        <svg
          class=""
          color="#EDEDEF"
          fill="currentColor"
          height="26"
          stroke="currentColor"
          stroke-width="0"
          style="color: rgb(237, 237, 239);"
          viewBox="0 0 24 24"
          width="26"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g>
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M12 2C6.5 2 2 6.5 2 12s4.5 10 10 10 10-4.5 10-10S17.55 2 12 2zm3.75 14.65c-2.35-1.45-5.3-1.75-8.8-.95-.35.1-.65-.15-.75-.45-.1-.35.15-.65.45-.75 3.8-.85 7.1-.5 9.7 1.1.35.15.4.55.25.85-.2.3-.55.4-.85.2zm1-2.7c-2.7-1.65-6.8-2.15-9.95-1.15-.4.1-.85-.1-.95-.5-.1-.4.1-.85.5-.95 3.65-1.1 8.15-.55 11.25 1.35.3.15.45.65.2 1s-.7.5-1.05.25zM6.3 9.75c-.5.15-1-.15-1.15-.6-.15-.5.15-1 .6-1.15 3.55-1.05 9.4-.85 13.1 1.35.45.25.6.85.35 1.3-.25.35-.85.5-1.3.25C14.7 9 9.35 8.8 6.3 9.75z"
              fill-rule="nonzero"
            />
          </g>
        </svg>
        <span>
          Spotify: 
          <span
            class="c28"
          >
            Not Playing
          </span>
        </span>
      </div>
      <menu
        class="c29"
      >
        <section>
          <h3
            class="c30"
          >
            Navigation
          </h3>
          <nav
            class="c31"
          >
            <a
              class="c32"
              href="/"
            >
              Home
            </a>
            <a
              class="c32"
              href="/about"
            >
              About
            </a>
            <a
              class="c32"
              href="/writings"
            >
              Writings
            </a>
            <a
              class="c32"
              href="/guestbook"
            >
              Guestbook
            </a>
          </nav>
        </section>
        <section>
          <h3
            class="c30"
          >
            Social
          </h3>
          <nav
            class="c31"
          >
            <a
              class="c32"
              href="https://github.com/doougui"
              target="_blank"
            >
              GitHub
            </a>
            <a
              class="c32"
              href="https://www.linkedin.com/in/douglaspigoulart/"
              target="_blank"
            >
              Linkedin
            </a>
            <a
              class="c32"
              href="https://twitter.com/oDougui"
              target="_blank"
            >
              Twitter
            </a>
            <a
              class="c32"
              href="mailto:douglaspigoulart@gmail.com"
              target="_blank"
            >
              E-mail
            </a>
          </nav>
        </section>
        <section>
          <h3
            class="c30"
          >
            Extras
          </h3>
          <nav
            class="c31"
          >
            <a
              class="c32"
              href="/"
              target="_blank"
            >
              Newsletter
            </a>
            <a
              class="c32"
              href="/"
              target="_blank"
            >
              Resume
            </a>
            <a
              class="c32"
              href="https://gitlab.com/douglaspigoulart"
              target="_blank"
            >
              GitLab
            </a>
          </nav>
        </section>
        <section
          class="c33"
        >
          <h3
            class="c30"
          >
            Newsletter
          </h3>
          <p
            class="c34"
          >
            Get new articles delivered to your inbox!
          </p>
          <div
            class="c35"
          >
            <input
              class="c36 c37"
              placeholder="johndoe@email.com"
            />
            <button
              class="c38"
            >
              Subscribe
            </button>
          </div>
        </section>
      </menu>
      <p
        class="c39"
      >
        © 2022 Douglas Pinheiro Goulart
      </p>
    </footer>
  </div>
</div>
`;
